File: ems-frontend/src/Services/EmployeeService.js
import axios from "axios";

const REST_API_BASE_URL = 'http://localhost:8080/api/employees';

//to get employees
export const listEmployees = () => axios.get(REST_API_BASE_URL);

//to add employees
export const createEmployee = (employee) => axios.post(REST_API_BASE_URL, employee);

//to get employee by id
export const getEmployee = (employeeId) => axios.get(REST_API_BASE_URL + '/' + employeeId);

//to update employee
export const updateEmployee = (employeeId, employee) => axios.put(REST_API_BASE_URL + '/' +employeeId, employee);

//to delete employee
export const deleteEmployee = (employeeId) => axios.get(REST_API_BASE_URL + '/' + employeeId);

File: ems-backend/ems-backend/src/main/java/com/example/webapp/ems_backend/service/EmployeeService.java
package com.example.webapp.ems_backend.service;

import com.example.webapp.ems_backend.dto.EmployeeDto;

import java.util.List;

public interface EmployeeService {

    EmployeeDto createEmployee(EmployeeDto employeeDto);//Create(POST) Employee

    EmployeeDto getEmployeeById(Long employeeId); //GET Employee

    List<EmployeeDto> getAllEmplyoees(); //GET All Employees

    EmployeeDto updateEmployee(Long employeeId, EmployeeDto updateEmployee); //PUT (update) Employee

    void deleteEmployee(Long employeeId); //DELETE Employee
}
